src/compat/posix/sys/mman/mprotect.c

```diff
@@ -0,0 +1,38 @@
+/**
+ * @file mprotect.c
+ * @brief Set protection of memory mapping
+ * @author Archit Checker <archit.checker5@gmail.com>
+ * @version
+ * @date 24.03.2020
+ */
+
+#include <errno.h>
+#include <stddef.h>
+
+#include <sys/mman.h>
+#include <hal/mmu.h>
+#include <mem/vmem.h>
+#include <mem/mmap.h>
+#include <kernel/task/resource/mmap.h>
+
+int mprotect(void* addr, size_t len, int prot) {
+	int err;
+	struct emmap *emmap = task_self_resource_mmap();
+
+	if ((int) addr % VMEM_PAGE_SIZE != 0) {
+		err = EINVAL;
+		return SET_ERRNO(err);
+	}
+
+	prot |= prot & (PROT_READ | PROT_WRITE | PROT_EXEC);
```

Comment:
It should be:

```
prot &= PROT_READ | PROT_WRITE | PROT_EXEC;
```

So we clear any other fields except these three ones.

Reply:
Corrected.